tls/certificate_store.go

```diff
@@ -2,14 +2,19 @@ package tls
 
 import (
 	"crypto/tls"
+	"sort"
+	"strings"
 
 	"github.com/containous/traefik/safe"
 )
 
 // CertificateStore store for dynamic and static certificates
 type CertificateStore struct {
-	DynamicCerts *safe.Safe
-	StaticCerts  *safe.Safe
+	DynamicCerts       *safe.Safe
+	StaticCerts        *safe.Safe
+	DefaultCertificate *tls.Certificate
+	CertCache          map[string]*tls.Certificate
```

Comment:
use a `map` as don't seem to be a good idea:
- what's happen when you update a dynamic cert?
- what's happen when an ACME cert is renew?